packages/controller/src/resources/cortex/index.ts

```diff
@@ -310,9 +317,30 @@ export function CortexResources(
     // limits_config
     // https://cortexmetrics.io/docs/configuration/configuration-file/#limits-config
     limits: {
+      // Delete blocks containing samples older than the specified retention
+      // period. Default is 0 (disabled). Add one day as a buffer here to be
+      // on the safe side. Note(JP): should this be the primary mechanism to
+      // delete old data from the corresponding cloud storage bucket? Should
+      // this replace the bucket lifecycle-based method?
+      compactor_blocks_retention_period: `${(metricRetentionDays + 1) * 24}h`,
```

Comment:
Nice! This is [pretty recent](https://github.com/cortexproject/cortex/pull/3879) and it doesn't even [show up in the docs yet](https://cortexmetrics.io/docs/blocks-storage/compactor/#compactor_config). I guess we have to test it to be sure?

Reply:
this part of the "compactor config" is actually documented in the `limits_config` which I was going through : - )

https://cortexmetrics.io/docs/configuration/configuration-file/#limits_config